src/commands/git.rs

```diff
@@ -722,6 +737,26 @@ fn cmd_git_push(
                 )?;
             }
         }
+
+        for commit in revision_commits {
```

Comment:
Perhaps this should warn if some of the revisions specified in `-r` were neither empty nor pushed? 

The warning could simply list the unpushed revisions or count them if there are many. Or, the warning could be a bit fancier and only mention the tips and says something like "Non-empty revisions tuv, xyz and 3 of their ancestors weren't pushed because no branches point to them. Either create branches pointing to them or use jj git push -c to push them with automatically names branches"

If you don't feel like it but think it's a good idea, you can leave a TODO.

Reply:
I think it would be a normal case for them to not be pushed. I would like to be able to do `jj git push -r main..@` even if most commits in the range don't have branches. It's probably a good idea to fail if there are no commits at all to push as a result of the `-r` argument(s), however. I made it so.

Reply:
I'm thinking of the case where the tree looks like this:

```
@ empty
|
o non-empty-commit-without-a-branch
|
o commit1  branch1
|
o commit2
|
o commit3  main
```

The fact that `non-empty-commit-without-a-branch` is not pushed on `jj git push -r main..@` could be surprising and/or indicate a mistake, so we could warn about that.

Meanwhile, the fact that `commit2` doesn't have a branch pointing right to it is fine, since it's covered by `branch1`. There's no need to say anything about such commits. Nor is there a need to warn about the empty commit.

Reply:
I think I would personally not want a warning about that case, especially not if we make `jj git push` without arguments push `remote_branches()..@` (or if I personally have a config or alias that does that). Since it's extra work to add a such a warning, I'm tempted to not add it until we hear if it's confusing. If we do add it, I suspect we actually should warn also about the empty head commit since the user included it in the revset. What do you think?

Reply:
The main thing I'm trying to prevent is the user forgetting `jj amend` or `jj branch s branch1 -r @-` and pushing less than they intended. For the former case, it's important to *not* warn about empty revisions. For the latter, it's important to warn about nonempty commits. So, I *would* want this warning even if I usually push `remote_branches()..@`.

It is possible I'm missing some other forest for these trees, though. It's also possible people will use this feature differently than I would. If so, we can make the warning configurable.

In any case, I don't think we should block this PR on deciding when and if we print warnings.

Reply:
Ah, that makes sense. We actually have warning for hg users at Google about pushing with a dirty working copy. I've disabled that warning myself, but I *think* most users find it useful and leave it enabled. In any case, I'll merge this now so we can try it for a while to see how it works for different users.